jobs/mysql_backup_job.go

```diff
@@ -96,7 +116,14 @@ func (j *MySQLBackupJob) Run(ctx context.Context, send Send) (rerr error) {
 		}
 	}
 
-	xtrabackupCmd := exec.CommandContext(ctx, xtrabackupBin, "--compress", "--backup") // #nosec G204
+	return nil
+}
+
+func (j *MySQLBackupJob) backup(pipeCtx context.Context) (rerr error) {
+	pipeCtx, cancel := context.WithCancel(pipeCtx)
+	defer cancel()
```

Comment:
The main idea is to kill all commands in the pipeline when one of them is failed.

Reply:
Also `cmd.Wait` order matters. Imagine situation where you have `cmd1 | cmd2`, so in your code you should with cmd2 completion and then cmd1.